src/main/java/org/ccci/gto/persistence/aopalliance/DeadLockRetryMethodInterceptorFactory.java

```diff
@@ -0,0 +1,22 @@
+package org.ccci.gto.persistence.aopalliance;
+
+import static org.ccci.gto.persistence.DeadLockRetry.DEFAULT_ATTEMPTS;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.ccci.gto.persistence.DeadLockRetryAspectSupport;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DeadLockRetryMethodInterceptorFactory extends DeadLockRetryAspectSupport {
+    private final Map<String, MethodInterceptor> advice = new HashMap<>();
+
+    public MethodInterceptor getAdvice(@Nonnull final String unitName) {
+        return advice.computeIfAbsent(unitName, this::createAdvice);
+    }
```

Comment:
who is going to call this method?

Reply:
this gets called within The Key's logic here:
https://github.com/CruGlobal/thekey/pull/264/files#diff-9248e2839976813a9e76e8af81d251b9R46

Reply:
it's meant to be used when you want to wrap functionality with deadlock retry logic, but can't modify the source to include a `@DeadLockRetry` annotation (e.g. stock CAS functionality).